src/LaMEM_post_processing.jl

```diff
@@ -0,0 +1,532 @@
+julia> model_name = "PEV"
+julia> timefile_location = "test/test_files/timestep/"
+julia> timestep = "Timestep_00000000_0.00000000e+00"
+julia> p_fields = ["phase", "temperature"]
+julia> surface_level = search_for_model_constrains(dat_path, "surf_level")
+julia> name = "track"*string(CartesianIndex(200,1,200))
+julia> output_dir = "test/test_files/timestep/"
+
+julia> track_point = track_point_over_time(Point_coord,p_fields,model_name,timefile_location,surface_level, name::String, output_dir::String,save = false)
+
+Dict{String, Dict{String, Float64}} with 2 entries:
+  "Timestep_00000010_1.49079279e+01" => Dict("phase"=>1.53445, "temperature"=>0.0)
+  "Timestep_00000000_0.00000000e+00" => Dict("phase"=>2.0, "temperature"=>0.0)
+   ...
+ ```
+"""
+
+
+
+# track one specific point over time for multiple fields
+
+function track_point_over_time(Point_coord::CartesianIndex,fields::Vector{String},model_name::String,timefile_location::String,surface_level::Vector, name::String, output_dir::String,save = false)
+
+    track_point = Dict{String,Dict{String,Float64}}()
+
+    time_files = filter(f -> startswith(f, "Time"), readdir(timefile_location))
+    save_dict = Dict{String,Dict{String,Dict{String,Float64}}}()
+
+    for timestep in time_files
+
+        data = get_data_timestep(timefile_location,timestep,FileName_pvtr,fields,surface_level,false)
+        track_point[timestep] = Dict{String,Float64}()
+
+        for field in fields
+
+            field_name = Symbol(field)
+            field_data = getfield(data.fields, field_name)
+
+            track_point[timestep][field] = field_data[Point_coord.I[1],Point_coord.I[2],Point_coord.I[3]]
+
+        end
+    end
+
+
+    if save == true
+        save_dict[model_name] = Dict{String,Dict{String,Float64}}()
+        save_dict[model_name] = track_point
+
+        file_name = string(name)*".txt"
+        output_name=joinpath(output_dir,file_name)
+
+        # Serialize the vector of structures to a file
+        open(output_name, "a") do file
+            serialize(file, save_dict)
+        end
+    end
+
+
+    return track_point
+
+end
+
+
+###################################################################################################################
+
+"""
+  file =   deserialize_file(output_dir::String,name::String)
+
+Parameters
+====
+- `output_dir` - directory of the file
+- `name` -  name of the file
+
+
+Examples
+========
+
+```julia
+
+julia> output_dir = "./output/
+julia> name = "track"*string(CartesianIndex(10,10,10))
+julia> file_info = deserialize_file(output_dir,name)
+
+1-element Vector{Any}:
+ Dict("VEP" => Dict("Timestep_00000010_1.49079279e+01" => Dict("phase" => 1.534447431564331, "temperature" => 0.0), 
+        "Timestep_00000000_0.00000000e+00" => Dict("phase" => 2.0, "temperature" => 0.0)))
+
+ ```
+"""
+
+
+
+# load and extract information of serialized files
+
+function deserialize_file(output_dir::String,name::String)
+
+    file_name = string(name)*".txt"
```

Comment:
```suggestion
    file_name = name * ".txt"
```
isnt `name` already a string?